Add categories to each changelog release

Gudahtt opened the pull request:
Each changelog release now has category headers. The standard ["keep a changelog"](https://keepachangelog.com/en/1.0.0/) categories are used, along with the addition of "Uncategorized" for any changes that have not yet been categorized.

The changelog script has been updated to add this "Uncategorized" header if it isn't already present, and to place any new commits under this header.

The changelog has been updated to property categorize each change in recent releases, and to place changes in older releases under the header "Uncategorized".

Relates to #10752

Manual testing steps:  

For development changelog updates:
- Run `yarn update-changelog`
- See that it places changelog entries under 'Unreleased`, under the "Uncategorized" header

For RC changelog updates with existing release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Add a new release header to the changelog for the new version
- Run `yarn update-changelog`
- See that it places changelog entries under the new release header under the "Uncategorized" category
 
For updates without a release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Run `yarn update-changelog`
- See that it places changelog entries under the newly added release header under the "Uncategorized" category

Gudahtt committed:
Add categories to each changelog release

Each changelog release now has category headers. The standard "keep a
changelog" [1] categories are used, along with the addition of
"Uncategorized" for any changes that have not yet been categorized.

The changelog script has been updated to add this "Uncategorized"
header if it isn't already present, and to place any new commits under
this header.

The changelog has been updated to property categorize each change in
recent releases, and to place changes in older releases under the
header "Uncategorized".

[1]: https://keepachangelog.com/en/1.0.0/

Gudahtt commented on development/auto-changelog.js:
```diff
@@ -69,6 +69,37 @@ async function main() {
   const changelog = await fs.readFile(changelogFilename, { encoding: 'utf8' });
   const changelogLines = changelog.split('\n');
 
+  const prNumbersWithChangelogEntries = [];
```
This was moved earlier in the script to allow the early exit on line 99, to ensure we don't add the 'Uncategorized' header to the "Unreleased" section if there are no changes.

As a bonus, it now avoids creating a new release header if there are no commits as well! Though that shouldn't happen generally.

Gudahtt committed:
Fix formatting mistakes in changelog

A few releases were missing the 'Uncategorized' header due to
inconsistencies in formatting. The formatting for each release has
been made consistent, and any missing 'Uncategorized' headers have
been added.

Gudahtt commented on CHANGELOG.md:
```diff
@@ -1844,42 +1861,40 @@ rollback to 3.10.0 due to bug
 - Add Parity compatibility, fixing Geth dependency issues.
-- Add a link to the transaction in history that goes to https://metamask.github.io/eth-tx-viz
-  too help visualize transactions and to where they are going.
+- Add a link to the transaction in history that goes to https://metamask.github.io/eth-tx-viz to help visualize transactions and to where they are going.
```
I moved this onto one line to make the formatting more consistent and easier to parse. I also fixed a typo here (`too` => `to`).

Gudahtt commented on CHANGELOG.md:
```diff
@@ -1844,42 +1861,40 @@ rollback to 3.10.0 due to bug
-- Fixed bug where if you send a transaction from within MetaMask extension the
-  popup notification opens up.
+### Uncategorized
+- Fixed bug where if you send a transaction from within MetaMask extension the popup notification opens up.
```
This was moved onto one line as well.

Gudahtt commented on CHANGELOG.md:
```diff
@@ -2149,18 +2164,18 @@ rollback to 3.10.0 due to bug
-
-Made seed word restoring BIP44 compatible.
+### Uncategorized
+- Made seed word restoring BIP44 compatible.
```
These older releases didn't have a `- ` in front of their changes. I've added it to make them consistent with the others.

brad-decker reviewed (dismissed):
This LGTM though I'd like some clarification on how the various categories are determined. What differentiates changed/added vs fixed?

Gudahtt commented:
The k[eep a changelog spec](https://keepachangelog.com/en/1.0.0/) describes the categories as follows:

- `Added` for new features.
- `Changed` for changes in existing functionality.
- `Deprecated` for soon-to-be removed features.
- `Removed` for now removed features.
- `Fixed` for any bug fixes.
- `Security` in case of vulnerabilities.

Sometimes the line between what counts as a change to an existing feature v.s a new feature is blurry v.s a bug fix is blurry. But that's fine.We can pick whichever we think would be most useful for readers.

Gudahtt commented on CHANGELOG.md:
```diff
@@ -1,35 +1,50 @@
+### Fixed
 - [#10692](https://github.com/MetaMask/metamask-extension/pull/10692): Prevent UI crash when a 'wallet_requestPermissions" confirmation is queued behind a "wallet_addEthereumChain" confirmation
 - [#10712](https://github.com/MetaMask/metamask-extension/pull/10712): Fix infinite spinner when request for token symbol fails while attempting an approve transaction
+- [#10486](https://github.com/MetaMask/metamask-extension/pull/10486): Add setting to hide zero balance tokens
```
What is this doing here :thinking: This was supposed to be in v9.2.0

Gudahtt replied:
Fixed in a7e6e05b90a96f8d89e3f1aa9f9b40b4fa45bb55

Gudahtt committed:
Fix miscategorized changelog entry

The "Add setting to hide zero balance tokens" changelog entry was
accidentally moved from v9.2.0 to v9.2.1. It has been moved back.
